---
title: "Barplot for blue's RNA-seq data in Human Brain"
author: "Shicheng Guo"
date: "August 17, 2016"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

**What's our life**:

![](https://lh3.googleusercontent.com/-EinFwWjANx0/V7S-jdr6TcI/AAAAAAAADrE/2rUG4IK8hCE_K8mW3x38xKb8szleSD3cQCJoC/w563-h325-p-rw/this.is.our.life.png)



## Example of usage on R markdown, rpubs, github and LabNotes wiki

**rpubs**: <https://rpubs.com/>

**github**: <https://github.com/>

**wiki**:<http://genome-tech.ucsd.edu/LabNotes/index.php/Shicheng_Guo>



Useful Script:
**R markdown cheat sheet** <https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf>

** R markdown slides** <http://rpubs.com/mansun_kuo/24330>


R -e 'library("rmarkdown");

library("knitr");

rmarkdown::render("NormalDevconJuly.Rmd")'

R -e 'library("markdown");

rpubsUpload("normalDev","NormalDevconJuly.html")'


library("knitr")

library("rmarkdown")

rmarkdown::render('DeconvolutionMixture.Rmd')

pandoc("DeconvolutionMixture.md",format="MediaWiki")

## Example 1
Here, I want to show how to use **R markdown**, **rpubs**, **github** and **LabNotes wiki** to Kun's lab member. I just use the bar plot example which I created yesterday for Blue. Blue told me he want to build a figure for his future manuscript, just like this one:

```{r, out.width = 800, fig.retina = NULL}
knitr::include_graphics("C:\\Users\\shicheng\\Documents\\GitHub\\Lab\\Cn39rtdUIAAi8n7.png")
```

Here is my code:

```{r blue}
setwd("C:\\Users\\shicheng\\Documents\\GitHub\\Lab")

barplotReAxis<-function(mp){
  mp<-c(0,mp)
  np<-c()
  end<-length(mp)-1
  for(i in 1:end){
    np.tmp<-(mp[i]+mp[i+1])/2
    np<-c(np,np.tmp)
  }
  np<-round(np)
  names(np)=names(mp)[2:length(mp)]
  return(np)
}


nbt.data<-data.matrix(read.table("BrainRNASeqBlue.txt",head=T,row.names=1,sep="\t",as.is=T))

nbt.data[1:5,1:5]

i=6
rmpk<-nbt.data[i,]
genename<-rownames(nbt.data)[i]
genename

cell.labels <- substr(names(rmpk),0,7)
cell.labels<-unlist(lapply(cell.labels,function(x) unlist(strsplit(x,"_"))[1]))
level=c(paste("In",1:8,sep=""),paste("Ex",1:8,sep=""))
group1<-factor(unlist(lapply(cell.labels,function(x) substr(x,1,3))))
group2<-factor(unlist(lapply(cell.labels,function(x) substr(x,4,nchar(x)))))
group3<-1:length(group2)
df<-data.frame(rmpk,cell.labels,group1,group2,group3)
df<-df[order(df$cell.labels),]
head(df)
```

**Now we are going to focusn on gene: `r genename`**

Okay, the above code is use to collect the data frame. Now, let's do the bar plot.

```{r,fig.width=10, fig.height=2.5}
filename=paste(genename,".png",sep="")
size=0.6
# png(filename,width = 8, height = 2, units = 'in', res = 300)
par(mar=c(2, 3, 1, 3), xpd=TRUE)
space<-rep(0,nrow(df))
space[cumsum(table(df$cell.labels))+1]<-30
space[cumsum(table(df$group1))+1]<-200
space<-space[1:(length(space)-1)]
bar.positions <- barplot(df$rmpk,border=df$group2,xaxt='n',space=space,main=genename,cex.main=0.6,cex.lab=size,cex.axis=size)
title(ylab = "Counts per gene", cex.lab = size,line = 2)
label<-levels(df$group1)
table(df$group1)
pos<-bar.positions[cumsum(barplotReAxis(table(df$group1)))]
axis(side=1,at=pos,labels=label,tick=FALSE,las=2,cex.axis=size)
legend(x=max(bar.positions)+100,y=max(rmpk)/2,legend=levels(df$group2),bty="n",lty=1,bg="transparent",col=c(as.numeric(unique(df$group2))),cex=0.4,inset=c(-0.2,0))
# dev.off()
```
